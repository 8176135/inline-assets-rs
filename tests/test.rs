extern crate inline_assets;

#[cfg(test)]
use pretty_assertions::{assert_eq, assert_ne};

#[cfg(test)]
use insta::assert_snapshot;

#[test]
fn basic() {

	let res = inline_assets::inline_file(
		"examples/resources/listener_screen.html",
		inline_assets::Config {
			remove_new_lines: false,
			inline_fonts: false
		}
	)
	.unwrap();
	println!("{}", res);
    assert_snapshot!(
        "listener_screen_basic",
		res
	);
}

#[test]
fn css_import_and_http() {
	let res = inline_assets::inline_file(
		"examples/resources/listener_screen_css_import.html",
		inline_assets::Config {
			remove_new_lines: false,
			inline_fonts: false,
		}
	)
	.unwrap();
	println!("{}", res);
	assert_snapshot!(
		"listener_screen_css_import",
		res
	);
}

//#[test]
//fn remove_new_lines() {
//	assert_eq!(
//		inline_assets::inline_file(
//			"examples/resources/listener_screen_css_import.html",
//			Default::default()
//		)
//		.unwrap(),
//		include_str!("listener_remove_new_lines.compiled.html")
//	);
//}
