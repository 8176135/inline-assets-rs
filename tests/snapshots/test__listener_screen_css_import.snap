---
source: tests/test.rs
assertion_line: 38
expression: res

---
<html><head>
    <style>
@font-face {
    font-family: 'source-sans-black';
    src: url(another_sub_dir/../SourceSansPro-Black.otf) format("woff");
}

body {
    background-image: url(https://i.imgur.com/SoqAxJR.png);
}

@font-face {
    font-family: 'source-sans';
    src: url(another_sub_dir/sourcesanspro-regular-webfont.woff2) format("woff");
}

body {
    font-family: "source-sans", serif;
    font-size: 10pt;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;

    background-color: #CCC;
    color: #222;
}

p {
    cursor: default;
}

#port_num {
    width: 50px;
}

button {
    width: auto;
    left: 5px;
    right: 5px;
    bottom: 5px;
    position: absolute;
    border: none;
    background-color: #7d81ff;
    color: #EEE;
}

button:hover {
    background-color: #a9a7ff
}

input, label {
    display: block;
}

#is_connected {
    margin-left: 20px;
    font-family: Consolas, monospace;
}</style>
</head>
<body>
<p style="overflow:hidden; max-height: 43px"><b>Status: </b> <span id="is_connected">Not listening</span></p>
<form>
    <label for="port_num">Port:</label>
    <input id="port_num" max="65500" min="1" name="port_num" placeholder="13111" type="number">
    <br><br>
    <button onclick="external.invoke('listen ' + valOrPlaceholder(document.getElementById('port_num')))" type="submit">Listen</button>
</form>
<script>function valOrPlaceholder(item) {
    if (item.value) {
        return item.value;
    } else {
        return item.placeholder;
    }
}</script>
<script type="text/javascript">
    function showConnectedMsg(e) {
        document.getElementById('is_connected').innerHTML = e;
    }
</script>

</body></html>
